source/validations.ts

```diff
@@ -172,6 +172,25 @@ export class Validations {
 		})
 	}
 
+	/**
+	 * Warns the user that the behaviour for `max: 0` is changing in the next
+	 * major release.
+	 *
+	 * @param max {number} - The maximum number of hits per client.
+	 *
+	 * @returns {void}
+	 */
+	// TODO: Remove in v7.0.0
+	max(max: number) {
+		this.wrap(() => {
+			if (max === 0) {
+				console.warn(
+					`The behaviour shown when 'max' is set to zero will change in v7.0.0. Please see https://express-rate-limit.github.io/WRN_MAX/ for more information on this change.`,
```

Comment:
Lets make the code `ERL_WRN_MAX_ZERO`

Also, might as well describe the old and new behavior briefly here. Maybe something like this:
```suggestion
					`Setting max to 0 disables rate limiting in in express-rate-limit 6 and older, but will cause all requests to be blocked in v7. See https://express-rate-limit.github.io/ERL_WRN_MAX_ZERO/ for more information on this change.`,
```

Finally, I think it does make sense to different breaking change warnings from other validation errors, but I kind of like the plumbing we have in place that gets a stack trace and formats the URL and whatnot. Perhaps we make a new `ChangeWarning extends ValidationError {}` and throw that, and then the try/catch can do console.error for validation errors and console.warn for change warnings? I'm not sure that's the best answer, but I think it's worth considering a bit more.

Reply:
Oh, and it doesn't have to be part of the error message, but in the wiki we can suggest that users create a custom `skip()` function to replicate the behavior of setting max to 0 without getting warned.

Reply:
Okay!